### PR TITLE
Implementing `_mm_shuffle_epi8`.

### LINKED ISSUE #17: Implement SSSE3's `_mm_shuffle_epi8`
Just like clang, we can use the llvm intrinsics `llvm.x86.ssse3.pshuf.b.128`.

### PR DESCRIPTION
**Pull request by fulmicoton**
Closes #17.

### CODE REVIEW

File: src/x86/ssse3.rs [hunk 1]
```diff
@@ -8,10 +8,45 @@ pub fn _mm_abs_epi8(a: i8x16) -> u8x16 {
     unsafe { pabsb128(a) }
 }
 
+
+#[inline(always)]
+#[target_feature = "+ssse3"]
```
Comment (BurntSushi) [inline src/x86/ssse3.rs]: Please follow the style in the rest of the code. :-) Put the attributes after the comments.

File: src/x86/ssse3.rs [hunk 2]
```diff
@@ -8,10 +8,45 @@ pub fn _mm_abs_epi8(a: i8x16) -> u8x16 {
...
+/// Shuffle bytes from `a` according to the content of `b`.
+///
+/// The last 4 bits of each bytes of `b` are used as as many
```
Comment (BurntSushi) [inline src/x86/ssse3.rs]: "... of each byte of ..."

Comment (BurntSushi) [reply src/x86/ssse3.rs]: "... are used as addresses into ..."

File: src/x86/ssse3.rs [hunk 3]
```diff
@@ -8,10 +8,45 @@ pub fn _mm_abs_epi8(a: i8x16) -> u8x16 {
...
+/// addresses into the 16 bytes of `a`.
+///
+/// In addition, if the highest significant bit of a byte of `b`
+/// is set, the respective destination byte is set to 0.
+///
+/// Picturing `a` and `b` as `[u8; 16]`, `_mm_shuffle_epi8` is
+/// logically equivalent to :
```
Comment (BurntSushi) [inline src/x86/ssse3.rs]: "... logically equivalent to: ..."

File: src/x86/ssse3.rs [hunk 4]
```diff
@@ -8,10 +8,45 @@ pub fn _mm_abs_epi8(a: i8x16) -> u8x16 {
...
+///     for i in 0..16 {
+///         // if the most significant bit of b is set,
+///         // then the destination byte is set to 0.
+///         if b[i] & 0x80 == 0u8 {
+///             r[i] = a[(b[i] % 16) as usize];
+///         }
+///     }
+///     r
+/// }
+/// ```
```
Comment (BurntSushi) [inline src/x86/ssse3.rs]: Awesome, I love using examples like this. :-)

**Comment by BurntSushi** [comment]
Awesome, thank you!

**Comment by fulmicoton** [comment]
Awesome!

**Pull request merged**